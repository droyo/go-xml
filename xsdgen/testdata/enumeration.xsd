<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
  xmlns="http://example.org/" targetNamespace="http://example.org/">

  <xs:simpleType name="dType">
    <xs:annotation>
      <xs:documentation>type for weekday</xs:documentation>
    </xs:annotation>
    <xs:restriction base="xs:string">
      <xs:enumeration value="MON"/>
      <xs:enumeration value="TUE"/>
      <xs:enumeration value="WED"/>
      <xs:enumeration value="THU"/>
      <xs:enumeration value="FRI"/>
      <xs:enumeration value="SAT"/>
      <xs:enumeration value="SUN"/>
    </xs:restriction>
  </xs:simpleType>

  <xs:simpleType name="intType">
    <xs:annotation>
      <xs:documentation>integer enum</xs:documentation>
    </xs:annotation>
    <xs:restriction base="xs:integer">
      <xs:enumeration value="1"/>
      <xs:enumeration value="2"/>
      <xs:enumeration value="3"/>
    </xs:restriction>
  </xs:simpleType>

  <xs:simpleType name="floatType">
    <xs:annotation>
      <xs:documentation>float enum</xs:documentation>
    </xs:annotation>
    <xs:restriction base="xs:decimal">
      <xs:enumeration value="0.9"/>
      <xs:enumeration value="0.333333333"/>
      <xs:enumeration value="0.0"/>
    </xs:restriction>
  </xs:simpleType>

</xs:schema>