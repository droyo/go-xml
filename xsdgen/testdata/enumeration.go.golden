// Code generated by xsdgen.test. DO NOT EDIT.

package ws

// May be one of MON, TUE, WED, THU, FRI, SAT, SUN
type DType string

const (
	DType_MON DType = "MON"
	DType_TUE DType = "TUE"
	DType_WED DType = "WED"
	DType_THU DType = "THU"
	DType_FRI DType = "FRI"
	DType_SAT DType = "SAT"
	DType_SUN DType = "SUN"
)

// May be one of 0.9, 0.333333333, 0.0
type FloatType float64

const (
	FloatType_0_9         FloatType = 0.9
	FloatType_0_333333333 FloatType = 0.333333333
	FloatType_0_0         FloatType = 0.0
)

// May be one of 1, 2, 3
type IntType int

const (
	IntType_1 IntType = 1
	IntType_2 IntType = 2
	IntType_3 IntType = 3
)
